### PR TITLE
Syncs project selector on login events

### LINKED ISSUE #1688: Cloud Storage window needs to be re-opened after login
1) Log out of your cloud account
2) Open the GCS bucket panel and note that buckets will not be displayed
3) Log in to the cloud account
4) Refresh the GCS buckets

-> The buckets won't refresh. Only upon minimizing and maximizing the panel will they be loaded.

### PR DESCRIPTION
**Pull request by etanshaul**
fixes #1688

Note for the full experience to be correct, #1693 needs to be fixed. I tested this with da734da stashed to make sure things work as expected.

This does 2 things:
1) Adds a message in the UI in the bucket listing panel when the user is logged out:
image
2) Fixes all the project selector synchronization issues by adding a login/logout listener and re-syncing the project selector on these events.

The listener mechanism for login events that was already in place wasn't robust enough. There was no way to to register the listener extension properly: 
- If I regestered it in `plugin.xml` it would fail because the listener needs to be an inner class of `ProjectSelector` to have access to it.
- If I programmatically register the extension in the `ProjectSelector`'s constructor, this would not work because the account plugin is loaded before the cloud tools plugin.

So instead I moved it to use the built in IntelliJ message bus pub/sub mechanism.

### CODE REVIEW

**Comment by etanshaul** [comment]
chasing a "Too many projects leaked" error in the test framework again

**Comment by etanshaul** [comment]
I _think_ the tests are fixed now.

@nkibler | @patflynn | @nbashirbello PTAL

File: google-account-plugin/testSrc/com/google/cloud/tools/intellij/login/CredentialedUserRosterTest.java [hunk 1]
```diff
@@ -47,8 +43,8 @@ public void setUp() {
   }
 
   /**
-   * Tests that {@link com.google.intellij.login.Users#addUser(CredentialedUser)}
-   * stores the proper users and currently manages the active user.
+   * Tests that {@link CredentialedUserRoster#addUser(CredentialedUser)} stores the proper users and
```
Comment (nkibler) [inline google-account-plugin/testSrc/com/google/cloud/tools/intellij/login/CredentialedUserRosterTest.java]: Are these comments necessary? Typically there are no javadoc comments on tests; it should be easily readable so the reader knows what it's testing.

File: google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties [hunk 2]
```diff
@@ -346,6 +346,7 @@ gcs.panel.bucket.listing.no.project.selected=To view your buckets select a Cloud
 gcs.panel.bucket.listing.error.loading.buckets=Could not load buckets for selected project.
 gcs.panel.bucket.listing.no.buckets.found=No buckets found for selected project.
 gcs.panel.bucket.listing.loading.text=loading ...
+gcs.panel.bucket.listing.not.logged.in=To view your buckets login to a Cloud Project.
```
Comment (nkibler) [inline google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: Technically you're logging into your Google account to view your Cloud projects. Maybe reword this?

File: google-account-plugin/testSrc/com/google/cloud/tools/intellij/login/CredentialedUserRosterTest.java (hunk 1 above)
Comment (etanshaul) [reply google-account-plugin/testSrc/com/google/cloud/tools/intellij/login/CredentialedUserRosterTest.java]: true. I wasn't planning on refactoring the existing code except for the broken reference. but I agree these are redundant.

File: google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties (hunk 2 above)
Comment (etanshaul) [reply google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: ah good point. how about: "... login to your Google Cloud Platform account" ?

Comment (etanshaul) [reply google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: ill check how we word this elsewhere too

Comment (nkibler) [reply google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: SGTM, but maybe split "log in" since "login" is a noun for the act of logging in. :)
